### PR TITLE
Display api errors on catching exception

### PR DESCRIPTION
**Pull request by gauravmagicbell**
Opening this PR to resolve
https://magicbell.productboard.com/roadmap/2566106-release-roadmap/features/7496637/detail/expanded
The PR does the following two things
1. adds api errors to MagicBell Error
2. displays the error suggestions
image

### CODE REVIEW

File: lib/magicbell/api_operations.rb [hunk 1]
```diff
@@ -34,11 +36,19 @@ def default_headers
...
       e = MagicBell::Client::HTTPError.new
       e.response_status = response.code
       e.response_headers = response.headers.to_h
       e.response_body = response.body
+      unless e.response_body.nil? || e.response_body.empty?
+        body = JSON.parse(response.body)
+        e.errors = body["errors"]
+        e.errors.each do |error, index|
+          puts "#{error["suggestion"].red}"
+          puts "#{error["help_link"].blue.on_white}"
```
Comment (unamashana) [inline lib/magicbell/api_operations.rb:49]: The colors are a nice touch :)

File: lib/magicbell/api_operations.rb [hunk 2]
```diff
@@ -34,11 +36,19 @@ def default_headers
...
       return if response.success?
-
+      
       e = MagicBell::Client::HTTPError.new
       e.response_status = response.code
       e.response_headers = response.headers.to_h
       e.response_body = response.body
+      unless e.response_body.nil? || e.response_body.empty?
+        body = JSON.parse(response.body)
+        e.errors = body["errors"]
```
Comment (unamashana) [inline lib/magicbell/api_operations.rb:46]: Would it also be better to have specific error classes? What are the pros and cons?

File: lib/magicbell/client.rb [hunk 3]
```diff
@@ -3,7 +3,8 @@ class Client
     class HTTPError < StandardError
       attr_accessor :response_status,
                     :response_headers,
-                    :response_body
+                    :response_body,
```
Comment (unamashana) [inline lib/magicbell/client.rb:6]: Are you going to add some specs for these?

File: lib/magicbell/api_operations.rb (hunk 2 above)
Comment (gauravmagicbell) [reply lib/magicbell/api_operations.rb:46]: Not sure if we should have specific error classes;
If we have specific error classes, everytime we change / add new error code in the api, the ruby client will have to be updated as well.

File: lib/magicbell/client.rb (hunk 3 above)
Comment (gauravmagicbell) [reply lib/magicbell/client.rb:6]: added

File: lib/magicbell/api_operations.rb (hunk 2 above)
Comment (unamashana) [reply lib/magicbell/api_operations.rb:47]: @gauravmagicbell I was thinking that the specific error classes help people do better error handling in their code. We don't have to work on them now but I think we should soon. 

We can always default to the more generic error classes if we don't have a specific one.

**Review by unamashana — APPROVED** [review]

**Pull request merged**